### PR TITLE
converting build-self-signed-jwt guide to single page

### PR DESCRIPTION
**Pull request by chrisdmills-okta**
## Description:
- **What's changed?** This PR converts https://developer.okta.com/docs/guides/build-self-signed-jwt/java/overview/ into a single-page guide. 
- **Is this PR related to a Monolith release?** No
- Redirects tested on staging, 11/21/21. They all work fine.

### Resolves:

* [OKTA-438737](https://oktainc.atlassian.net/browse/OKTA-438737)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/build-self-signed-jwt/main/index.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+---
+
+**Learning outcomes**
+
+* Build a signed JWT for use when making requests to the OpenID Connect endpoints
+* Understand which claims are required in your JWT payload
+* Sign the JWT with a shared secret or a private key
+
+**What you need**
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/build-self-signed-jwt/main/index.md:19]:
```suggestion
```

removing this section based on nut-facts template

File: packages/@okta/vuepress-site/docs/guides/build-self-signed-jwt/main/index.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+n/a
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/build-self-signed-jwt/main/index.md:21]:
```suggestion
```

File: packages/@okta/vuepress-site/docs/guides/build-self-signed-jwt/main/index.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+| iss      | Required. The issuer of the token. This value must be the same as the `client_id` of the application that you are accessing.  | String      |
+| sub      | Required. The subject of the token. This value must be the same as the `client_id` of the application that you are accessing. | String       |
+| jti      | Optional. The unique token identifier. If you specify this parameter, the token can only be used once and, as a result, subsequent token requests won't succeed. | String    |
+| iat      | Optional. When the token was issued in seconds since January 1, 1970 UTC (UNIX timestamp), for example, `1555591219`. If specified, it must be a time before the request is received. | Integer     |
+
+## Build a JWT with a shared key
+
+If you configured your application to use the `client_secret_jwt` client authentication method, then you want to build a JWT that you sign with the `client_secret` using an HMAC SHA algorithm (HS256, HS384, or HS512).
+
+<StackSelector snippet="createclientsecretjwt"/>
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/build-self-signed-jwt/main/index.md:61]: Do you want to put this stack selector before the nut-facts based on the new template?
Add `noSelector` here if you are moving the selector up top.

File: packages/@okta/vuepress-site/docs/guides/build-self-signed-jwt/main/index.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+## Build a JWT with a private key
+
+If you configured your client to use the `private_key_jwt` client authentication method, then you want to build a JWT that you sign with your private key using an RSA or ECDSA algorithm (RS256, RS384, RS512, ES256, ES384, ES512).
+
+<StackSelector snippet="createprivatekeyjwt"/>
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/build-self-signed-jwt/main/index.md:67]:
```suggestion
<StackSelector snippet="createprivatekeyjwt" noSelector />
```

File: packages/@okta/vuepress-site/conductor.yml [hunk 5]
```diff
@@ -3475,6 +3475,26 @@ redirects:
...
+  - from: /docs/guides/build-self-signed-jwt/-/overview/index.html
+    to: /docs/guides/build-self-signed-jwt/
+  - from: /docs/guides/build-self-signed-jwt/-/gather-information/index.html
+    to: /docs/guides/build-self-signed-jwt/-/main/#gather-claims-information
+  - from: /docs/guides/build-self-signed-jwt/-/jwt-with-shared-key/index.html
+    to: /docs/guides/build-self-signed-jwt/-/main/#build-a-jwt-with-a-shared-key
+  - from: /docs/guides/build-self-signed-jwt/-/jwt-with-private-key/index.html
+    to: /docs/guides/build-self-signed-jwt/-/main/#build-a-jwt-with-a-private-key
+  - from: /docs/guides/build-self-signed-jwt/-/next-steps/index.html
+    to: /docs/guides/build-self-signed-jwt/
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/conductor.yml:3497]: Based on our previous discussion about stack-selector, unfortunately, you have to add another set of redirects for each language in the original guide.

**Review by vanngo-okta — COMMENTED** [review]
The set of redirects look good. Unfortunately, I think we need to add redirects for each language supported in the original guide?
Added a few comments about updated nutrition fact template.

File: packages/@okta/vuepress-site/docs/guides/build-self-signed-jwt/main/index.md (hunk 3 above)
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/build-self-signed-jwt/main/index.md:57]: Yup, I'll add a visible stack selector up top, and hide all the others. Thanks!

File: packages/@okta/vuepress-site/conductor.yml (hunk 5 above)
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/conductor.yml:3497]: Ah yes, I'll get that one done too. This was one of those that I did before I realized we had to do that ;-)

**Comment by chrisdmills-okta** [comment]
@vanngo-okta ok, updates made, and redirects tested on staging.

**Review by vanngo-okta — APPROVED** [review]
LGTM 🚢

**Comment by chrisdmills-okta** [comment]
@vanngo-okta woo hoo, let's do this!

**Pull request merged**